README.md

```diff
@@ -167,18 +167,22 @@ You can add options to your code to `allow`/`warn`/`deny` Clippy lints:
 
 *   `allow`/`warn`/`deny` can be limited to a single function or module using `#[allow(...)]`, etc.
 
-Note: `deny` produces errors instead of warnings.
+Note: `allow` in this case means to "allow your code to have the lint without
+warning".  `deny` means "produce an error if your code has the lint".  `warn`
+means "produce a warning, but don't produce an error due to this lint".  An
+error causes clippy to exit with an error code, so is useful in scripts like
+CI/CD.
```

Comment:
I don't like the wording here. Code cannot "have a lint". But a lint triggers for code. Also just because something is marked with `warn`, doesn't mean the lint will trigger.

> `allow` means to suppress the lint for your code. With `warn` the lint will only emit a warning, while with `deny` the lint will emit an error, when triggering for your code. An error causes clippy to exit with an error code, so is useful in scripts like CI/CD.

Reply:
Thanks for taking a look!

I like your reword, I was trying to figure out the right wording but couldn't come up with anything better than "having lint on a shirt" terminology.  I've pushed the change!